test/integration/compliance_history_test.go

```diff
@@ -194,86 +221,258 @@ var _ = Describe("GRC: [P1][Sev1][policy-grc] Test the compliance history API",
 		)
 		Expect(err).ToNot(HaveOccurred())
 
-		var parentPolicy *unstructured.Unstructured
-		var clusters []string
+		DeferCleanup(func(ctx context.Context) {
+			By("Deleting the policy")
+			_, err := common.OcHub(
+				"delete",
+				"-f",
+				"../resources/compliance_history/policy.yaml",
+				"--ignore-not-found",
+			)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		_, clusters := verifyPolicyOnAllClusters(ctx, policyNS, policyName, "NonCompliant", defaultTimeoutSeconds*2)
 
+		expectedEvents := len(clusters) * 2
+
+		By(fmt.Sprintf("Verifying that there are %d compliance events for the parent policy", expectedEvents))
 		Eventually(func(g Gomega) {
-			By("Checking to see if the policy is noncompliant on all clusters")
-			var err error
-			clusters = []string{}
+			for _, cluster := range clusters {
+				// event.timestamp_after is used to filter compliance events from previous runs if the test
+				// is run multiple times. This won't be needed after https://issues.redhat.com/browse/ACM-9314 is
+				// addressed.
+				events, err := listComplianceEvents(
+					ctx, "cluster.name="+cluster, "parent_policy.name="+policyName, "event.timestamp_after="+now,
+				)
+				g.Expect(err).ToNot(HaveOccurred())
+
+				g.Expect(events).To(HaveLen(2), fmt.Sprintf("expected cluster %s to have two events", cluster))
+
+				event1 := events[0].(map[string]interface{})
+				event2 := events[1].(map[string]interface{})
+
+				if event1["policy"].(map[string]interface{})["name"].(string) == "default-namespace-must-exist" {
+					g.Expect(event1["event"].(map[string]interface{})["compliance"]).To(Equal("Compliant"))
+					g.Expect(event2["event"].(map[string]interface{})["compliance"]).To(Equal("NonCompliant"))
+				} else {
+					g.Expect(event1["event"].(map[string]interface{})["compliance"]).To(Equal("NonCompliant"))
+					g.Expect(event2["event"].(map[string]interface{})["compliance"]).To(Equal("Compliant"))
+				}
+			}
+		}, defaultTimeoutSeconds, 1).Should(Succeed())
+	})
+
+	It("Creates a policy with a Gatekeeper constraint", func(ctx context.Context) {
+		const installGKPolicyName = "compliance-api-install-gk"
+		const uninstallGKPolicyName = "compliance-api-uninstall-gk"
+		const gkTargetNS = "compliance-api-test"
+		const invalidConfigMapName = "compliance-api-test"
+		const gkPolicyName = "compliance-api-gk"
+
+		By("Creating the " + installGKPolicyName + " policy to install Gatekeeper")
+		_, err := common.OcHub(
+			"apply",
+			"-f",
+			"../resources/compliance_history/policy-install-gk.yaml",
+		)
+		Expect(err).ToNot(HaveOccurred())
 
-			parentPolicy, err = clientHubDynamic.Resource(common.GvrPolicy).Namespace(policyNS).Get(
-				ctx, policyName, metav1.GetOptions{},
+		DeferCleanup(func(ctx context.Context) {
```

Comment:
DeferCleanup doesn't work when `It` failed right? so I wrapped with Describe and used `afterall`

Reply:
I haven't noticed that issue. It looks like the docs say that it's the same as an AfterEach under the hood:
https://onsi.github.io/ginkgo/#cleaning-up-our-cleanup-code-defercleanup